### PR TITLE
DAF-5565: Disable PIP and Notification logic

### PR DESCRIPTION
**Pull request by mvn-hieunguyen-hn**
## Description
- Release notification and hide PIP control button

### Solution
- Release notification and hide PIP control button

### What was done (Please be a little bit specific)
- Release notification and hide PIP control button

### JIRA
https://dw-ml-nfc.atlassian.net/browse/DAF-5565

## Screenshot or Video (Before/After)
Added later in PR on the dwango-app-ch repo.

## Ready for review checklist
Basically, you must check all. (When you ignore any checks, please write each reason.)

- [x] All sections above are properly filled in
- [x] The PR only deals with one functionality/bug
- [ ] Includes code refactoring? If yes, the following sub checks are required.
    - [ ] The changes were tested. So there is no degradation.
    - [ ] The change is not too big. (Reviewers can review it.)
- [ ] Includes change of spec? If yes, the sub check is required.
    - [ ] Updated the document.
- [x] Builds and runs on iOS (No new warnings nor new errors)
- [x] Builds and runs on Android (No new warnings nor new errors)

### CODE REVIEW

File: ios/Classes/BetterPlayerPlugin.m [hunk 1]
```diff
@@ -627,6 +626,23 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
...
+
+            if (!player.isPipMode) {
+                [player resetPipController];
+            } else {
+                [_notificationPlayer setIsDisplayPipButtons:false];
+            }
+            [[NSNotificationCenter defaultCenter] removeObserver:self];
+            [self disposeNotificationData:player];
+            [self removeCommandCenterTargetHandlers];
+            [self setRemoteCommandsNotificationNotActive];
```
Comment (mvn-thanhluu-hn) [inline ios/Classes/BetterPlayerPlugin.m]:
```suggestion
            [[UIApplication sharedApplication] endReceivingRemoteControlEvents];
```
In this function have disable `AVAudioSession`, I think we only need to use  `endReceivingRemoteControlEvents`. Please check it.

File: ios/Classes/BetterPlayerPlugin.m [hunk 2]
```diff
@@ -627,6 +626,23 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
...
+            [self itemDidPlayToEndTime:nil];
```
Comment (mvn-thanhluu-hn) [inline ios/Classes/BetterPlayerPlugin.m]: We disposed notification data in L641, so I think we don't need it.

**Review by mvn-thanhluu-hn — COMMENTED** [review]
@mvn-hieunguyen-hn 
Thanks. 

I have some comments. Please check it 🙏 .

**Review by mvn-thanhluu-hn — APPROVED** [review]
LGTM 👍 

@noboru-i 
Please review this PR 🙏

**Review by noboru-i — COMMENTED** [review]
Sorry, I think, we already handle "Disable PIP" by calling `setupAutomaticPictureInPictureTransition`.

Can't we use it in this case?
Which point did we have a difference?

**Comment by mvn-hieunguyen-hn** [comment]
@noboru-i 
I tried using `setupAutomaticPictureInPictureTransition` but when I call it in PIP mode, it doesn't work as expected.
So I had to try this way.

File: ios/Classes/BetterPlayerPlugin.m [hunk 3]
```diff
@@ -627,6 +626,22 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
                     [player showPlayerCoverView];
                 }
             }
+        } else if([@"hidePipAndControlCenterButton" isEqualToString:call.method]) {
+            NSDictionary* dataSource = [_dataSourceDict objectForKey:[self getTextureId:player]];
+            NSMutableDictionary *mutableDataSource = [dataSource mutableCopy];
+            [mutableDataSource setObject:@(false) forKey:@"showNotification"];
+            _dataSourceDict[[self getTextureId:player]] = mutableDataSource;
+
+            if (!player.isPipMode) {
```
Comment (noboru-i) [inline ios/Classes/BetterPlayerPlugin.m]: It is difficult to understand.
Please swap condition and process.

```
            if (player.isPipMode) {
                [_notificationPlayer setIsDisplayPipButtons:false];
            } else {
                [player resetPipController];
            }
```

**Review by noboru-i — COMMENTED** [review]
I add a small comment.
Please check it 🙏

**Review by noboru-i — APPROVED** [review]
LGTM 👍

**Pull request merged**